lib/web/desktop/playback.go

```diff
@@ -125,163 +52,107 @@ const (
 // control playback.
 type actionMessage struct {
 	Action        playbackAction `json:"action"`
-	PlaybackSpeed float32        `json:"speed,omitempty"`
-}
-
-// waitWhilePaused waits idly while the player's state is paused, waiting until:
-// - the play state is toggled back to playing
-// - the play state is set to finished (the player is closed)
-func (pp *Player) waitWhilePaused() {
-	pp.cond.L.Lock()
-	defer pp.cond.L.Unlock()
-
-	for pp.playState == playStatePaused {
-		pp.cond.Wait()
-	}
-}
-
-// togglePlaying toggles the state of the player between playing and paused,
-// and wakes up any goroutines waiting in waitWhilePaused.
-func (pp *Player) togglePlaying() {
-	pp.cond.L.Lock()
-	defer pp.cond.L.Unlock()
-	switch pp.playState {
-	case playStatePlaying:
-		pp.playState = playStatePaused
-	case playStatePaused:
-		pp.playState = playStatePlaying
-	}
-	pp.cond.Broadcast()
+	PlaybackSpeed float64        `json:"speed,omitempty"`
 }
 
-// close closes the websocket connection, wakes up any goroutines waiting on the playState condition,
-// and cancels the playbackPlayer's context.
-//
-// It should be deferred by all the goroutines that use playbackPlayer,
-// in order to ensure that when one goroutine closes, all the others do too.
-func (pp *Player) close(cancel context.CancelFunc) {
-	pp.closeOnce.Do(func() {
-		pp.mu.Lock()
-		defer pp.mu.Unlock()
+// ReceivePlaybackActions handles logic for receiving playbackAction messages
+// over the websocket and updating the player state accordingly.
+func ReceivePlaybackActions(
+	ctx context.Context,
+	log logrus.FieldLogger,
+	ws *websocket.Conn,
+	player *player.Player) {
+	// assume playing by default
+	// (this is because the player in the web UI implements a single play/pause action,
+	// but the underlying player API has separate play and pause methods)
```

Comment:
Nit: I found this parenthetical more confusing than clarifying, I think something like this would suffice

```suggestion
	// Playback always starts in Play mode
```